### ISSUE TITLE
Query-Result has more Docs than the Collection itself

### ISSUE THREAD

**Issue by lanluu** [arangodb/arangodb#4843]
Hi, we are running into a trouble with the following query

`
FOR order IN orders FILTER order.TableTurnId == "TT_-L7Ub0jRiPJksFf4UptU" 
AND order.AppId == "2c86f85f-8753-4c95-9001-ecd779ada73c"
RETURN DISTINCT order
`

The _orders_ Collection contains only 5 Docs. The given query returns 2 Docs, as we expect. If we remove the **DISTINCT** keyword, 74 Docs were returned/delivered. Almost were duplicated, with the same _rev.

We tried many variants and figured out that the GUILD-based string _order.AppId_ caused. The index for it is "hash". 
We also tried LIKE instead of "==" for the _order.AppId_, it worked absolutely nice, as we expect. (edited)

----
**Query-Execution-Explanation** 
Query string:
 FOR order IN orders
 FILTER 
 order.TableTurnId == "TT_-L7Ub0jRiPJksFf4UptU" 
 AND order.AppId == "2c86f85f-8753-4c95-9001-ecd779ada73c"
 RETURN order
Execution plan:
 Id  NodeType    Est.  Comment
 1  SingletonNode   1  * ROOT
 6  IndexNode     1   - FOR order IN orders  /* hash index scan */
 5  ReturnNode     1    - RETURN order
Indexes used:
 By  Type  Collection  Unique  Sparse  Selectivity  Fields                                               Ranges
 6  hash  orders    false  false    99.96 %  [ `AppId`, `TableTurnId`, `Items[*].ItemId`, `Items[*].MenuItemVariantId`, `Items[*].Category` ]  ((order.`AppId` == "2c86f85f-8753-4c95-9001-ecd779ada73c") && (order.`TableTurnId` == "TT_-L7Ub0jRiPJksFf4UptU"))
Optimization rules applied:
 Id  RuleName
 1  use-indexes
 2  remove-filter-covered-by-index
 3  remove-unnecessary-calculations-2

-----
simran.arangodb:
your index is a combined index over multiple fields including arrays, which are likely causing the result
However, indexes should generally not affect the query result, but only improve performance.
Please create an issue on Github so that it can be tracked

**Comment by jsteemann** [comment]
Reproduce code (can be used regardless of storage engine):
```js
db._create("orders");
db.orders.insert({ AppId: "2c86f85f-8753-4c95-9001-ecd779ada73c", TableTurnId: "TT_-L7Ub0jRiPJksFf4UptU", Items: [{Category:1, MenuItemVariantId: 1}, {Category:1, MenuItemVariantId: 2 }, { Category:2, MenuItemVariantId: 1 }, { Category: 2, MenuItemVariantId: 2 } ]});
db.orders.ensureIndex({ type: "skiplist", fields: ["AppId", "TableTurnId", "Items[*].ItemId", "Items[*].MenuItemVariantId", "Items[*].Category"] });
q = `FOR order IN orders FILTER order.TableTurnId == "TT_-L7Ub0jRiPJksFf4UptU" AND order.AppId == "2c86f85f-8753-4c95-9001-ecd779ada73c" RETURN order`; 
db._query(q).toArray().length;
```
This returns `4` although there is only one document present in the collection. Looks like there needs to be some extra deduplication in the array index result generation.

**Comment by jsteemann** [comment]
The issue is caused by using multiple expanded (`[*]`) attributes in the same index. The result is as expected with just a single expanded attribute.

**Comment by jsteemann** [comment]
Will be fixed by https://github.com/arangodb/arangodb/pull/4854

**Comment by OmarAyo** [comment]
Hi @lanluu  ,

Thanks for reporting this bug

We have released 3.3.5, and it includes a fix for it
I am closing this ticket now. Please feel free to reopen if the problem persists
